lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -327,32 +326,17 @@ def draw(self, renderer):
         marker_transform = marker_scale + marker_rotation
 
         for loc, angle in self.locs_angles:
-            marker_rotation.rotate_deg(angle+add_angle)
-            locs = path_trans.transform_non_affine(np.array([loc, loc]))
+            marker_rotation.clear().rotate_deg(angle+add_angle)
+            locs = path_trans.transform_non_affine([loc])
+            if (self.axes and
+                    not self.axes.viewLim.contains(locs[0][0], locs[0][1])):
```

Comment:
@QuLogic I realize this is a bit old, but I think this check is not correct?  You've applied part of path_trans (only the non-affine part) to `loc` at this point, so if path_trans (i.e. self.get_transform()) is not linear the containment check may be not at all what we need?

I don't have a real failing case though (as I'm not sure how to get a log-scaled axisartist...) and am not even sure what the correct containment check would be, though.

Reply:
Possibly, but without a specific case, I guess I also don't know whether it's really wrong or how to fix it. It certainly fixed _something_, as indicated in the commit message.

I'm not entirely sure if it's the right test, but I applied:
```patch
diff --git a/examples/axisartist/demo_parasite_axes.py b/examples/axisartist/demo_parasite_axes.py
index d3fe972d4..52066a59c 100644
--- a/examples/axisartist/demo_parasite_axes.py
+++ b/examples/axisartist/demo_parasite_axes.py
@@ -24,8 +24,11 @@ import matplotlib.pyplot as plt
 fig = plt.figure()
 
 host = HostAxes(fig, [0.15, 0.1, 0.65, 0.8])
+host.set_yscale('log')
 par1 = ParasiteAxes(host, sharex=host)
+par1.set_yscale('log')
 par2 = ParasiteAxes(host, sharex=host)
+par2.set_yscale('log')
 host.parasites.append(par1)
 host.parasites.append(par2)
 
@@ -44,8 +47,8 @@ p2, = par1.plot([0, 1, 2], [0, 3, 2], label="Temperature")
 p3, = par2.plot([0, 1, 2], [50, 30, 15], label="Velocity")
 
 host.set_xlim(0, 2)
-host.set_ylim(0, 2)
-par1.set_ylim(0, 4)
+host.set_ylim(0.001, 2)
+par1.set_ylim(0.001, 4)
 par2.set_ylim(1, 65)
 
 host.set_xlabel("Distance")
```
and it does appear that ticks are visible and that this code is hit.

Reply:
OK, let's not bother for now then.